DigitalLearningSolutions.Data.Tests/Services/CourseAdminFieldsServiceTests.cs

```diff
@@ -74,21 +78,103 @@ public void GetCustomPromptsWithAnswersForCourse_Returns_Populated_List_of_Custo
+            A.CallTo(() => courseAdminFieldsDataService.UpdateCustomPromptForCourse(1, 1, null)).DoesNothing();
 
             // When
-            courseAdminFieldsService.UpdateCustomPromptForCourse(1, 1, true, null);
+            courseAdminFieldsService.UpdateCustomPromptForCourse(1, 1, null);
 
             // Then
-            A.CallTo(() => courseAdminFieldsDataService.UpdateCustomPromptForCourse(1, 1, true, null))
-                .MustHaveHappened();
+            A.CallTo(() => courseAdminFieldsDataService.UpdateCustomPromptForCourse(1, 1, null)).MustHaveHappened();
+        }
+
+        [Test]
+        public void GetCoursePromptsAlphabeticalList_calls_data_service()
+        {
+            // Given
+            const string promptName = "Access Permissions";
+            A.CallTo(() => courseAdminFieldsDataService.GetCoursePromptsAlphabetical()).Returns
+                (new List<(int, string)> { (1, promptName) });
+            // When
+            var result = courseAdminFieldsService.GetCoursePromptsAlphabeticalList();
+
+            // Then
+            A.CallTo(() => courseAdminFieldsDataService.GetCoursePromptsAlphabetical()).MustHaveHappened();
+            result.Contains((1, promptName)).Should().BeTrue();
+        }
+
+        [Test]
+        public void AddCustomPromptToCourse_adds_prompt_to_course_at_next_prompt_number()
+        {
+            // Given
+            A.CallTo
+            (
+                () => courseAdminFieldsDataService.UpdateCustomPromptForCourse(100, A<int>._, A<int>._, null)
+            ).DoesNothing();
+            A.CallTo(() => courseAdminFieldsDataService.GetCourseAdminFields(100, 101, 0))
+                .Returns(CustomPromptsTestHelper.GetDefaultCourseAdminFieldsResult());
+
+            // When
+            var result = courseAdminFieldsService.AddCustomPromptToCourse(100, 101, 0, 3, null);
+
+            // Then
+            A.CallTo
+            (
+                () => courseAdminFieldsDataService.UpdateCustomPromptForCourse(100, 3, 3, null)
+            ).MustHaveHappened();
+            result.Should().BeTrue();
+        }
+
+        [Test]
+        public void AddCustomPromptToCourse_does_not_add_prompt_if_course_has_all_prompts_defined()
+        {
+            // Given
+            A.CallTo
+            (
+                () => courseAdminFieldsDataService.UpdateCustomPromptForCourse(100, A<int>._, A<int>._, null)
+            ).DoesNothing();
+            A.CallTo(() => courseAdminFieldsDataService.GetCourseAdminFields(100, 101, 0))
+                .Returns(
+                    CustomPromptsTestHelper.GetDefaultCourseAdminFieldsResult(
+                        "System Access Granted",
+                        "Test",
+                        false,
+                        "Priority Access",
+                        "",
+                        false,
+                        "Access Permissions"

```

Comment:
Could we just pass in three default prompts to this helper to make this more intuitive? (so change the parameter types of CustomPromptsTestHelper.GetDefaultCourseAdminFieldsResult)

Reply:
We could, but then this helper will be constructed differently from the GetDefaultCentreCustomPromptsResult helper. Is that okay or should I change them both?

Reply:
Ah, now that I look at it again, it's not too bad actually, especially when we remove the mandatory fields so I'd leave it 👍